analytics_data_api/management/commands/generate_stage_course_data.py

```diff
@@ -0,0 +1,64 @@
+# pylint: disable=line-too-long,invalid-name
+
+import datetime
+import logging
+
+from analytics_data_api.management.commands.generate_data import FakeDataCommand
+from analytics_data_api.v0 import models
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class Command(FakeDataCommand):
+    help = 'Generate fake data for stage environment'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--database',
+            action='store',
+            dest='database',
+            default='default',
+            help='Database in which to generate fake date',
+        )
+
+    def get_start_date(self, database):
+        enrollments = models.CourseEnrollmentDaily.objects.using(database)
+        # default to four weeks of data if no data exists
+        start_date = datetime.datetime.now() - datetime.timedelta(weeks=4)
+        if enrollments:
+            latest_date = enrollments.latest().date
+            # want to return latest date + some time difference to avoid overlap
+            start_date = datetime.datetime.combine(
+                latest_date + datetime.timedelta(days=1), datetime.datetime.min.time()
+            )
+        return start_date
+
+    def handle(self, *args, **options):
+        course_ids = ['course-v1:edX+DemoX+Demo_Course']
+        insights_course_ids = ['course-v1:edX+Insights+Stage', 'course-v1:edX+Insights+Fake+Data']
+        course_ids += insights_course_ids
+        usernames = ['ed_xavier', 'xander_x', 'alice_bob']
+        program_title = 'edX Insights'
+        database = options['database']
+
+        logger.info("Option to generate videos with ids pulled from the LMS is disabled, using fake video ids...")
+        video_ids = self.fake_video_ids_fallback()
+
+        start_date = self.get_start_date(database)
+        end_date = datetime.datetime.now()
+
+        for course_id in course_ids:
+            logger.info("Generating data for %s in database %s", course_id, database)
+            self.generate_weekly_data(course_id, start_date, end_date, database, delete_data=False)
+            self.generate_daily_data(
+                course_id, start_date, end_date, database,
+                delete_data=False, add_birth_year=False, use_current_cumulative=True
+            )
+            self.generate_all_video_data(course_id, video_ids, database)
+            for username in usernames:
+                self.generate_learner_engagement_data(course_id, username, start_date, end_date, database)
+            self.generate_learner_engagement_range_data(course_id, start_date.date(), end_date.date(), database)
+            self.generate_tags_distribution_data(course_id, database)
+
+        self.generate_program_data(insights_course_ids, program_title, database)
```

Comment:
is this supposed to be insights_course_ids rather than course_ids?

Reply:
Yes, wanted to only have a subset of the courses be in a program so we could check that filtering by program worked.